### PR TITLE
feat(material/select): add page down/up button functionality

### PR DESCRIPTION
**Pull request by forsti0506**
Hello,

small PR on adding some functionality according to (https://github.com/angular/components/issues/3549) and implemented to fit the standard.

- Page-Down focuses the last element in the select
- Page-Up focus the first element in the select
- It's done in open and closed state (https://w3c.github.io/aria-practices/examples/combobox/combobox-select-only.html there it is only in closed state, but for example in Chrome a native select is working cloed and open this way)

Best regards,

Martin

### CODE REVIEW

File: src/cdk/a11y/key-manager/list-key-manager.ts [hunk 1]
```diff
@@ -280,6 +282,22 @@ export class ListKeyManager<T extends ListKeyManagerOption> {
           return;
         }
 
+      case PAGE_UP:
+        if (isModifierAllowed) {
+          this.setFirstItemActive();
```
Comment (crisbeto) [inline src/cdk/a11y/key-manager/list-key-manager.ts:287]: Page up/down aren't supposed to focus the first/last items, but rather move one page up/down in the list. The reason why we haven't implemented it in the `ListKeyManager` is that it doesn't know how large the list is and the offset at which focus is set currently.

Comment (forsti0506) [reply src/cdk/a11y/key-manager/list-key-manager.ts:287]: Hi, if you take a look at my W3C link, it is possible to go 10 elements down/up or select the first/last option for the implentation of page_up/page_down. Or did I udnerstand somethin wrong from this reference implementation?

Comment (crisbeto) [reply src/cdk/a11y/key-manager/list-key-manager.ts:287]: The number 10 seems a little arbitrary to me, my understanding is that at least in native applications, it's supposed to go up/down by a page (hence the name).

Comment (forsti0506) [reply src/cdk/a11y/key-manager/list-key-manager.ts:287]: I definitly understand your concerns. I think the idea behing is, that a combobox is an "popup" which is after opening handled like it is a own page => therefore the jump of 10 (or last/first).

If you like I can implement the jump for 10 items, or we can forget the PR. I only saw it is an easy fix and thought i contribute it :)

Comment (crisbeto) [reply src/cdk/a11y/key-manager/list-key-manager.ts:287]: I think that's fine, but we should make it configurable, similar to how it's done for the home/end keys. E.g. something with `ListKeyManager.withPageUpDown(10)`.

Comment (forsti0506) [reply src/cdk/a11y/key-manager/list-key-manager.ts]: I pushed some tests and the integration with jumping 10 or to first/last one! Are there sth missing?

File: src/cdk/a11y/key-manager/list-key-manager.ts [hunk 2]
```diff
@@ -280,6 +294,25 @@ export class ListKeyManager<T extends ListKeyManagerOption> {
           return;
         }
 
+      case PAGE_UP:
+        if (this._pageUpAndDown.enabled && isModifierAllowed) {
+          const targetIndex = this._activeItemIndex - this._pageUpAndDown.delta;
+          this._setActiveItemByIndex(targetIndex > 0 ? targetIndex : 0, 1);
```
Comment (crisbeto) [inline src/cdk/a11y/key-manager/list-key-manager.ts:300]: You should be able to simplify this and the PAGE_DOWN handling using the `_setActiveItemByDelta` method instead.

**Review by crisbeto — COMMENTED** [review]
The CI is failing, because:
1. The API goldens need to be updated. You can do it by running `yarn approve-api a11y` and committing the changed file.
2. There's a merge commit which is tripping up the commit message validator. Usually we rebase branches to avoid merge commits.

File: src/cdk/a11y/key-manager/list-key-manager.ts (hunk 2 above)
Comment (forsti0506) [reply src/cdk/a11y/key-manager/list-key-manager.ts:300]: During writing the code I first had the same intention, but in my opinion this method can only be used to move 1 up or down and not an defined number or did i misunderstand anything?

**Comment by forsti0506** [comment]
> The CI is failing, because:
> 
>     1. The API goldens need to be updated. You can do it by running `yarn approve-api a11y` and committing the changed file.
> 
>     2. There's a merge commit which is tripping up the commit message validator. Usually we rebase branches to avoid merge commits.

1. file is updated
2. Should I make the rebase with the current main branch or is it done short before the merge?

File: src/cdk/a11y/key-manager/list-key-manager.ts (hunk 2 above)
Comment (crisbeto) [reply src/cdk/a11y/key-manager/list-key-manager.ts:300]: I see. Let's keep it like this for now.

**Review by crisbeto — APPROVED** [review]
LGTM

**Comment by forsti0506** [comment]
> LGTM

Nice, cu soon hopefully :)

**Pull request merged**